tidy3d/components/source.py

```diff
@@ -465,6 +465,29 @@ def _dir_vector(self) -> Tuple[float, float, float]:
         return None
 
 
+class BroadbandSource(Source, ABC):
+    """A source with frequency dependent field distributions."""
+
+    num_freqs: pydantic.NonNegativeInt = pydantic.Field(
```

Comment:
Should this rather be `pydantic.PositiveInt`? What happens if the `num_freqs = 0`?